packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -86,9 +86,18 @@ abstract class WidgetsBindingObserver {
   /// [SystemChannels.navigation].
   ///
   /// The default implementation is to call the [didPushRoute] directly with the
-  /// [RouteInformation.location].
+  /// string constructed from [RouteInformation.uri]'s path and query parameters.
   Future<bool> didPushRouteInformation(RouteInformation routeInformation) {
-    return didPushRoute(routeInformation.location!);
+    final Uri uri = routeInformation.uri;
+    return didPushRoute(
+        Uri.decodeComponent(
+          Uri(
+            path: uri.path.isEmpty ? '/' : uri.path,
+            queryParameters: uri.queryParametersAll.isEmpty ? null : uri.queryParametersAll,
+            fragment: uri.fragment.isEmpty ? null : uri.fragment,
+          ).toString(),
+        ),
```

Comment:
The goal is that Android and iOS can start sending
`http://mydomain.com/mypath` instead of just `/mypath` when it is deeplinked without breaking the existing code.